runtime/interp.c

```diff
@@ -677,29 +736,36 @@ value caml_interprete(code_t prog, asize_t prog_size)
       mlsize_t i;
       value block;
       if (wosize <= Max_young_wosize) {
-        Alloc_small(block, wosize, tag);
+        Alloc_small(block, wosize, tag, Enter_gc);
         Field(block, 0) = accu;
         for (i = 1; i < wosize; i++) Field(block, i) = *sp++;
       } else {
         block = caml_alloc_shr(wosize, tag);
+        Setup_for_c_call;
```

Comment:
This change looks dubious. If the calling convention of `caml_initialize` has actually changed, it should be documented and the other calls (e.g., the opcodes `CLOSURE` and `CLOSUREREC` above) should be fixed accordingly.

Reply:
I agree with @silene .  `caml_initialize` cannot trigger a GC, neither in Multicore OCaml nor in the current trunk, so there's no need to save and restore state around calls to `caml_initialize`.

Reply:
This is a vestige of the old concurrent minor GC. Let me clean this up.

Reply:
Fixed in https://github.com/ocaml/ocaml/pull/10831/commits/e7a8ac1b8f95c7cd6dc837e1d37cfba0688b3b84.